modules/parquet/package.json

```diff
@@ -50,16 +51,21 @@
     "lzo": "^0.4.11",
     "node-int64": "^0.4.0",
     "object-stream": "0.0.1",
+    "parquet-wasm": "^0.3.1",
     "snappyjs": "^0.6.0",
     "thrift": "^0.14.2",
     "varint": "^5.0.0",
     "zstd-codec": "^0.1"
   },
+  "peerDependencies": {
+    "apache-arrow": "*"
```

Comment:
Does this really work with all versions of apache-arrow?

Reply:
The only `apache-arrow` APIs this PR uses are:

- `RecordBatchStreamWriter.writeAll` https://github.com/visgl/loaders.gl/blob/a745c404044d9dac3b9fd095c70b3a3fd78c2a09/modules/parquet/src/lib/wasm/encode-parquet-wasm.ts#L39
- `RecordBatchStreamReader.from` and `new Table`:
https://github.com/visgl/loaders.gl/blob/a745c404044d9dac3b9fd095c70b3a3fd78c2a09/modules/parquet/src/lib/wasm/parse-parquet-wasm.ts#L36-L41

Those have existed for a while. At least in `apache-arrow` v4 ([writer](https://github.com/apache/arrow/blob/f959141ece4d660bce5f7fa545befc0116a7db79/js/src/ipc/writer.ts#L301-L314), [reader](https://github.com/apache/arrow/blob/f959141ece4d660bce5f7fa545befc0116a7db79/js/src/ipc/reader.ts#L144-L152)), v7 ([reader](https://github.com/apache/arrow/blob/e90472e35b40f58b17d408438bb8de1641bfe6ef/js/src/ipc/reader.ts#L145-L166), [writer](https://github.com/apache/arrow/blob/e90472e35b40f58b17d408438bb8de1641bfe6ef/js/src/ipc/writer.ts#L98-L109)), v1 ([writer](https://github.com/apache/arrow/blob/b0d623957db820de4f1ff0a5ebd3e888194a48f0/js/src/ipc/writer.ts#L97-L108), [reader](https://github.com/apache/arrow/blob/b0d623957db820de4f1ff0a5ebd3e888194a48f0/js/src/ipc/reader.ts#L144-L165)).

Reply:
So given that at this point the only use of `apache-arrow` in this package is to convert to/from the arrow IPC format, it's probably not necessary to pin a more strict version 🤷‍♂️